lib/rouge/lexers/cake.rb

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    load_lexer 'csharp.rb'
+
+    class Cake < CSharp
+      tag 'cake'
+      aliases 'cake'
+      filenames '*.cake'
+      mimetypes 'text/x-cake'
+
+      title "Cake (C# Make)"
+      desc 'A cross platform build automation system with a C# DSL'
+
+      preprocessor_keywords = %w(
+        addin break l load r reference tool
+      )
+
+      reference_keywords = %w(
+        Task IsDependentOn RunTarget
+      )
+
+      keywords = %w(
+        Does WithCriteria
+        Argument EnvironmentVariable
+        ContinueOnError Finally OnError ReportError
+        Setup TaskSetup TaskTeardown Teardown
+      )
+
+      prepend :root do
+        rule /^#[ \t]*(#{preprocessor_keywords.join('|')})\b.*?\n/, Comment::Preproc
```

Comment:
(This would need to be changed in the CSharp lexer as well, that hasn't gotten as much love as it needs)